src/net/udp/mod.rs

```diff
@@ -376,29 +377,30 @@ impl fmt::Debug for UdpSocket {
 #[must_use = "futures do nothing unless polled"]
 pub struct SendDgram<T>(Option<(UdpSocket, T, SocketAddr)>);
 
-fn incomplete_write(reason: &str) -> io::Error {
-    io::Error::new(io::ErrorKind::Other, reason)
-}
-
 impl<T> Future for SendDgram<T>
     where T: AsRef<[u8]>,
 {
     type Item = (UdpSocket, T);
     type Error = io::Error;
 
     fn poll(&mut self) -> Poll<(UdpSocket, T), io::Error> {
-        {
-            let (ref sock, ref buf, ref addr) =
-                *self.0.as_ref().expect("SendDgram polled after completion");
```

Comment:
Could this be left as-is? By using `as_ref` it avoids traffic on moves by moving the data here in and out unnecessarily.